collapsing dataframes

vicfabienne opened the pull request:
changed previous procedure for collapsing the deconvolution-dataframe to use the new function "dedupe" now. Added procedure to collaps the rows in the matching signature mutation table

Commit:
prep for develop_mode

Commit:
prep for develop_mode

Commit:
change to development mode

Commit:
change to development mode

Commit:
switch magic 13 to proper column name

Commit:
switch magic 13 to proper column name

Commit:
adding function "dedupeMuts" for collapsing rows in sigmut dataframe

Commit:
adding function "dedupeMuts" for collapsing rows in sigmut dataframe

Commit:
split chunk "processa_and_match_snvs_to_signature_mutations" in chunk for processing the sigmut dataframe and filtering the variant dataframe

Commit:
split chunk "processa_and_match_snvs_to_signature_mutations" in chunk for processing the sigmut dataframe and filtering the variant dataframe

Commit:
adding procedure for deduplicating the sigmut dataframe and grouping the variant names per mutation

Commit:
adding procedure for deduplicating the sigmut dataframe and grouping the variant names per mutation

Commit:
change "left_join" so the variant dataframe is now joined with the deduplicated version of the sigmut dataframe

Commit:
change "left_join" so the variant dataframe is now joined with the deduplicated version of the sigmut dataframe

Commit:
fixed typo in line 312

Commit:
fixed typo in line 312

Commit:
change "getIdentCols" function to the refined function "dedupe"

Commit:
change "getIdentCols" function to the refined function "dedupe"

Commit:
adapting chunk "collapse_identicals" to use "dedupe" function

Commit:
adapting chunk "collapse_identicals" to use "dedupe" function

Commit:
adapting chunk "getting_unique_muts_bulk", match.df is already unique now and doesn't need that extra "distinct()" step anymore

Commit:
adapting chunk "getting_unique_muts_bulk", match.df is already unique now and doesn't need that extra "distinct()" step anymore

Commit:
fix typo in line 375

Commit:
fix typo in line 375

Commit:
comment development chunks again

rekado commented on scripts/variantreport_p_sample.rmd:
```diff
@@ -34,11 +34,11 @@ outFile <- params$outFile
-# vepfile <- '/mnt/beast/pathogenomics/pigx_sarscov2_ww/output/vep/report_ready.txt' 
-# snvfile <- '/mnt/beast/pathogenomics/vpipe/akalinlab_pathogenomics/samples/wastewater_all/Test/variants/SNVs/snvs.csv'
+# vepfile <- 'tests/sample_data/Test_vep_sarscov2_parsed.txt' 
+# snvfile <- 'tests/sample_data/Test_snv.csv'
```
Since these lines are commented could they instead be removed?

vicfabienne replied:
hm, I use them to be able to test directly if this script still works when I'm working on it. 
Would it be ok, if we remove them at a later stage when we have reasonable tests and testing procedures in place?

rekado replied:
Sure.

I guess we should take some time to make that happen sooner.  How does this report get its "real" arguments now?

rekado commented on scripts/variantreport_p_sample.rmd:
```diff
@@ -150,35 +150,19 @@ deconv <- function (bulk,sig){
+        names (deduped)[names (deduped) %in% grouped] <- groupName
+        return (deduped)
+    } else {
+        return (df)
```
I'm glad this turned out to be useful!  I bet we could make it slightly more efficient by removing all but one of the collapsed columns before applying `duplicated` for a second time.  This may be an important optimization in case the data frame becomes very large with *many* columns and *many* rows, as we could cut down on the number of column comparisons.

vicfabienne replied:
Btw: When I used this to develop the dedupMuts I realized that dedupe will not work when we have the case that we have different groups of identical columns. Because dedupe just merges everything which is duplicated in any place. 

I mean when I e.g. maybe would need column names like: "wt - b117 - b1351,p1 - b1651 - b1429,b1427" 
I think with dedupe we would end up in a "wt - b117 - b1351,p1,b1429,b1427 - b1651 - b1351,p1,b1429,b1427" which would be wrong.

rekado replied:
I'm having a hard time understanding the example column names; what's the separator and how many columns are there in both cases?  And why would it be wrong?  What would the correct column names look like?

vicfabienne replied:
hmm sorry. :( 

Ok considering the case: >> "wt" - "b117" - "b1351,p1" - "b1651" - "b1429,b1427" <<, 5 columns 
As a product from a DF with previously 7 columns. But "b1351 & p1" are identical and "b1429 & b1427" are identical. 
But "b1351,p1" are NOT identical with "b1429,b1427". 

In dedupe you use "duplicated" to identify the identical columns and then you merge all the column names for which we get a TRUE into one single group. That's why I think with this example instead of "b1351,p1" & "b1429,b1427" as two seperate column names, we only will get "b1351,p1,b1429,b1427" as a new column name.

vicfabienne replied:
But it's just what I observed when I tried to use it as it is for the sigmut dataframe. But maybe I just did smth else different there. 

I want and will investigate this further with some dummy data for the example I described above. But it works right now. This will be a problem later when we have more or different variants. But yes - fixing/double-checking this will be at the top of the list after we have smth deliverable till Friday.

rekado commented on scripts/variantreport_p_sample.rmd:
```diff
@@ -288,15 +272,32 @@ sig_mutations.df <- as.data.frame(qpcR:::cbind.na(b117=b117$AA_mut, b1351=b1351$
+# fixme: I was not sure, how I can include this in my function but should be possible, will come back to it later
+dupes <- duplicated (sig_mutations.df$value, fromLast=TRUE)
+allDupes <- dupes | duplicated (sig_mutations.df$value, fromLast=FALSE)
+
```
this looks like a partial copy of the `dedupe` procedure.

Could this be merged with `dedupe`?  Or should `dedupe` be split into two parts to get the list of duplicate columns with one procedure and deduplicate the data frame in another procedure?

vicfabienne replied:
I think that partially goes with my comment from above. 

But yes - I think splitting dedupe would be a good idea.

mirifax committed:
Dev rendersite (#11)

* added rendercite rules to snakemake

adapted snakefile and rendercite to communicate with report_scripts

* dummy Kraken and Krona output files - to test rendersite

updated timecourse page to be in nav_bar

* moved two report scripts of vic into report_scripts/

added qc-report and its params  to rendersite generation

* added rule to snakefile and from .rmd to .Rmd

* changed test file to new sample csv file

* modified _settings and suppressed warnings

added snakemake rules for rendersite and commented out the ones which are not needed any more

* renamed variantreport

inserted variantreport to communicate with snakefile and generaterenderFiles. Also added the rule to the snakefile

* changed test paths in files and added plot snippets from rafael

deletet vep file because old

changed settings file so timecourse can read in the params and added test_vep file

added bubble figure try

 deleted commented lines for pull request

* adapted st reports are generated via render_site

removed commented lines for pull request

* added krona to kraken report

* deleted krona report, this will be handled by kraken2_report

changed name of kraken2-report to taxonomic classification

added index.rmd

* changed paste0 to file.path()

Co-authored-by: Dohmen <jan.dohmen@googlemail.com>

mirifax committed:
adapting to renderStie structure

* it's still a mixed structure with some development-testing parts left

adapted snakefile and rendercite to communicate with report_scripts

* dummy Kraken and Krona output files - to test rendersite

updated timecourse page to be in nav_bar

* moved two report scripts of vic into report_scripts/

added qc-report and its params  to rendersite generation

* added rule to snakefile and from .rmd to .Rmd

* changed test file to new sample csv file

* modified _settings and suppressed warnings

added snakemake rules for rendersite and commented out the ones which are not needed any more

* renamed variantreport

inserted variantreport to communicate with snakefile and generaterenderFiles. Also added the rule to the snakefile

* changed test paths in files and added plot snippets from rafael

deletet vep file because old

changed settings file so timecourse can read in the params and added test_vep file

added bubble figure try

 deleted commented lines for pull request

* adapted st reports are generated via render_site

removed commented lines for pull request

* added krona to kraken report

* deleted krona report, this will be handled by kraken2_report

changed name of kraken2-report to taxonomic classification

added index.rmd

* changed paste0 to file.path()

Co-authored-by: Dohmen <jan.dohmen@googlemail.com>

mirifax committed:
adapting to renderStie structure

* it's still a mixed structure with some development-testing parts left

adapted snakefile and rendercite to communicate with report_scripts

* dummy Kraken and Krona output files - to test rendersite

updated timecourse page to be in nav_bar

* moved two report scripts of vic into report_scripts/

added qc-report and its params  to rendersite generation

* added rule to snakefile and from .rmd to .Rmd

* changed test file to new sample csv file

* modified _settings and suppressed warnings

added snakemake rules for rendersite and commented out the ones which are not needed any more

* renamed variantreport

inserted variantreport to communicate with snakefile and generaterenderFiles. Also added the rule to the snakefile

* changed test paths in files and added plot snippets from rafael

deletet vep file because old

changed settings file so timecourse can read in the params and added test_vep file

added bubble figure try

 deleted commented lines for pull request

* adapted st reports are generated via render_site

removed commented lines for pull request

* added krona to kraken report

* deleted krona report, this will be handled by kraken2_report

changed name of kraken2-report to taxonomic classification

added index.rmd

* changed paste0 to file.path()

Co-authored-by: Dohmen <jan.dohmen@googlemail.com>

Commit:
changing structure, adding metadata

Commit:
uncomment test env

Commit:
comment renderSite env

Commit:
comment renderSite env

Commit:
uncomment test env

Commit:
adding "dedupeMuts" again

I lost the function in the recent merging process

Commit:
adding "dedupeMuts" again

I lost the function in the recent merging process

Commit:
adding start of writing var_output

Commit:
adding start of writing var_output

Commit:
adding sample name var

Commit:
adding sample name var

Commit:
adding comment for input args

Commit:
adding comment for input args

Commit:
adding skeleton for output.csv

dataframe skeleton and first functions are added for creating the variant_output.

Commit:
adding skeleton for output.csv

dataframe skeleton and first functions are added for creating the variant_output.

Commit:
adding variable for csv output directory

Commit:
adding variable for csv output directory

Commit:
writing  data_variant_plot.csv

added part for writing the csv file which is used to plot the variant abundance.

Commit:
writing  data_variant_plot.csv

added part for writing the csv file which is used to plot the variant abundance.

Commit:
switching off/commenting dev chunks

Commit:
switching off/commenting dev chunks

Commit:
remove redundant "samplename"

samplename variable changed to sample_name since it was already defined earlier

Commit:
remove redundant "samplename"

samplename variable changed to sample_name since it was already defined earlier

Commit:
resolved conflicts

Commit:
adding csv-writing part again

Idk why but for some reason it got lost during the merging process

Commit:
adding csv-writing part again

Idk why but for some reason it got lost during the merging process

Commit:
Merge remote-tracking branch 'origin/fix_hardcoding_variantreport' into fix_hardcoding_variantreport

# Conflicts:
#	scripts/report_scripts/variantreport_p_sample.Rmd
